torch/csrc/jit/codegen/cuda/lower_expr_sort.cpp

```diff
@@ -764,36 +803,41 @@ ExprGroup* ExprSegmentationSorter::makeMergedNode(
   // Merge the compute at domain of all edges going out from the newly joined
   // group. The val's we're looking for are from our consumer edges, but we want
   // to grab the producer val as that's the one we generate.
-  std::vector<IterDomain*> joined_ca_domains;
+  std::vector<std::vector<IterDomain*>> ca_domains;
   for (auto consumer_group_edge : joined_groups->consumerEdges()) {
     auto producer_of_consumer_edge = consumer_group_edge->producer_val_;
     if (producer_of_consumer_edge->isA<TensorView>()) {
       auto tv = producer_of_consumer_edge->as<TensorView>();
-      std::vector<IterDomain*> local_ca_domains;
+      std::vector<IterDomain*> local_ca_domains(
+          tv->getComputeAtPosition(), nullptr);
       for (size_t tv_i = 0; tv_i < tv->getComputeAtPosition(); tv_i++) {
-        local_ca_domains.push_back(tv->axis(tv_i));
+        local_ca_domains[tv_i] = tv->axis(tv_i);
       }
-      joined_ca_domains = mergeDomains(joined_ca_domains, local_ca_domains);
+      ca_domains.push_back(local_ca_domains);
     }
   }
-  joined_groups->payload()->ca_domains_ = joined_ca_domains;
+  joined_groups->payload()->ca_domains_ =
+      mergeDomains(ca_domains, GpuLower::current()->caLoopMap());
```

Comment:
The new way of merging domains sort of makes sense, but what's the reasoning of preferring domain merging by looking at all CA domains rather than just a pair of CA domain sets?

Reply:
Needs to be entire subgraph, updating.